pkg/webhook/conversion/decoder.go

```diff
@@ -30,8 +30,11 @@ type Decoder struct {
 }
 
 // NewDecoder creates a Decoder given the runtime.Scheme
-func NewDecoder(scheme *runtime.Scheme) (*Decoder, error) {
-	return &Decoder{codecs: serializer.NewCodecFactory(scheme)}, nil
+func NewDecoder(scheme *runtime.Scheme) *Decoder {
+	if scheme == nil {
+		panic("scheme should never be nil")
```

Comment:
lol at the old code. But why panic rather than returning an err and keeping the signature?

Reply:
Given that NewDecoder should only be called internally, I opted for a panic (it also looks like there are other places that we check fields and panic)

I'm also fine just removing every panic honestly, if that's what we prefer?

Reply:
If it should only be called internally, it shouldn't be exported IMHO

Reply:
Well, it looks like our examples are using it, even though there the builder would achieve the same thing?

Reply:
Maybe change them to either deal with the erroring version or make them use the builder?

Reply:
Let's do this as a follow-up? A panic here is probably ok, given that scheme is a parameter, not sure what one would expect if you do NewDecoder(nil) 🤷

Reply:
Fixed the examples/builtin one, there are still ways that folks might want to use a decoder, let's keep it exported for now. We might want to revisit most of the webhook package at some point, it's really cumbersome and there is lots of repetition between admission/conversion/authentication which can be avoided. It could also be good to see if we can do something with generics that's a bit more elegant